diff -ur nmap-and.orig/liblua/llex.c nmap-and/liblua/llex.c
--- nmap-and.orig/liblua/llex.c	2011-02-15 22:09:22.142710542 +0100
+++ nmap-and/liblua/llex.c	2011-02-15 22:11:50.442710540 +0100
@@ -176,9 +176,13 @@
 
 static void trydecpoint (LexState *ls, SemInfo *seminfo) {
   /* format error: try to update decimal point separator */
-  struct lconv *cv = localeconv();
   char old = ls->decpoint;
+#ifdef __ANDROID__
+  ls->decpoint = '.';
+#else
+  struct lconv *cv = localeconv();
   ls->decpoint = (cv ? cv->decimal_point[0] : '.');
+#endif
   buffreplace(ls, old, ls->decpoint);  /* try updated decimal separator */
   if (!luaO_str2d(luaZ_buffer(ls->buff), &seminfo->r)) {
     /* format error with correct decimal point: no more options */
diff -ur nmap-and.orig/nping/nping.h nmap-and/nping/nping.h
--- nmap-and.orig/nping/nping.h	2011-02-15 22:10:13.502710541 +0100
+++ nmap-and/nping/nping.h	2011-02-15 22:14:35.934710551 +0100
@@ -112,7 +112,7 @@
     #endif /* WIN32 */
 #endif /* HAVE_CONFIG_H */
 
-#ifndef WIN32
+#ifndef __ANDROID__
     #include <sysexits.h>
 #endif
 
diff -urN nmap-5.61TEST4.orig//configure nmap-5.61TEST4/configure
 --- nmap-5.61TEST4.orig//configure	2011-12-19 06:16:20.000000000 +0100
 +++ nmap-5.61TEST4/configure	2012-01-10 06:52:40.189167371 +0100
 @@ -4494,7 +4494,7 @@
  
  needs_cpp_precomp=no
  
 -LUA_CFLAGS="-DLUA_USE_POSIX -DLUA_USE_DLOPEN"
 +LUA_CFLAGS="-DLUA_USE_POSIX"
  
  case "$host" in
    *alpha-dec-osf*)
